test/plugins/test-trace-http2.ts

```diff
@@ -0,0 +1,366 @@
+        s.on('error', () => {
+          endTransaction();
+          common.assertDurationCorrect(Date.now() - start);
+          const span = common.getMatchingSpan((span: TraceSpan) => {
+            return span.name !== 'outer';
+          });
+          assert.equal(
+              span.labels[TraceLabels.ERROR_DETAILS_NAME],
+              'Error [ERR_HTTP2_STREAM_ERROR]');
+          assert.equal(
+              span.labels[TraceLabels.ERROR_DETAILS_MESSAGE],
+              'Stream closed with error code 2');
+          session.destroy();
+          server.close();
+          done();
+        });
+        s.end();
+      });
+    });
+  });
+
+  it('should accurately measure request time, event emitter', (done) => {
+    server.listen(common.serverPort, () => {
+      common.runInTransaction((endTransaction: () => void) => {
+        const start = Date.now();
+        const session = http2.connect(`http://localhost:${common.serverPort}`);
+        const s: http2.ClientHttp2Stream = session.request({':path': '/'});
+        s.setEncoding('utf8');
+        s.on('response', () => {
+          let result = '';
+          s.on('data', (data) => {
+             result += data;
+           }).on('end', () => {
+            endTransaction();
+            assert.equal(result, common.serverRes);
+            common.assertDurationCorrect(Date.now() - start);
+            session.destroy();
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  it('should handle concurrent requests', function(done) {
+    this.timeout(10000);  // this test takes a long time
+    let count = 200;
+    const slowServer: http2.Http2Server = http2.createServer();
+    slowServer.on('stream', (s) => {
+      setTimeout(() => {
+        s.respond({':status': count++});
+        s.end();
+      }, 5000);
+    });
+    slowServer.listen(common.serverPort, () => {
+      common.runInTransaction((endTransaction: () => void) => {
+        let completed = 0;
+        for (let i = 0; i < 5; i++) {
+          const session =
+              http2.connect(`http://localhost:${common.serverPort}`);
+          const s = session.request({':path': '/'});
+          s.on('data', () => {}).on('end', () => {
+            if (++completed === 5) {
+              endTransaction();
+              const spans = common.getMatchingSpans((span: TraceSpan) => {
+                return span.name !== 'outer';
+              });
+              assert.equal(spans.length, 5);
+              // We need to check a property attached at the end of a span.
+              const statusCodes: number[] = [];
+              for (let j = 0; j < spans.length; j++) {
+                const code = Number(
+                    spans[j].labels[TraceLabels.HTTP_RESPONSE_CODE_LABEL_KEY]);
+                assert.equal(statusCodes.indexOf(code), -1);
+                statusCodes.push(code);
+              }
+              assert.equal(statusCodes.reduce((a, b) => a + b), 1010);
```

Comment:
Neat

Reply:
This code was just copied from `http` :). I also think it's cute though.